src/System.Text.Json/tests/Serialization/ExceptionTests.cs

```diff
@@ -114,6 +131,66 @@ public static void PathForDictionaryFails()
             }
         }
 
+        [Fact]
+        public static void DeserializePathForDictionaryFails()
+        {
+            const string Json = "{\"Key1\u0467\":1, \"Key2\u0467\":bad}";
+            const string JsonEscaped = "{\"Key1\\u0467\":1, \"Key2\\u0467\":bad}";
+            const string Expected = "$.Key2\u0467";
+
+            JsonException e;
+
+            // Without custom escaper.
+            e = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<Dictionary<string, int>>(Json));
+            Assert.Equal(Expected, e.Path);
+
+            e = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<Dictionary<string, int>>(JsonEscaped));
+            Assert.Equal(Expected, e.Path);
+
+            // Custom escaper should not change Path.
+            var options = new JsonSerializerOptions();
+            options.Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping;
+
+            e = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<Dictionary<string, int>>(Json, options));
+            Assert.Equal(Expected, e.Path);
+
+            e = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<Dictionary<string, int>>(JsonEscaped, options));
+            Assert.Equal(Expected, e.Path);
+        }
+
+        private class ClassWithUnicodePropertyName
+        {
+            public int Property\u04671 { get; set; } // contains a trailing "1"
+        }
+
+        [Fact]
+        public static void DeserializePathForObjectFails()
+        {
+            const string GoodJson = "{\"Property\u04671\":1}";
```

Comment:
Interesting. This is the same issue as https://github.com/dotnet/corefx/pull/40793#discussion_r320569279 where GitHub gets tripped up by const strings containing characters written out in their "escaped" format, where it starts to consider this as invalid C# (even though it isn't). Note, the string keyword isn't being syntax-highlighted correctly either.

Looks like it's caused by the C# property names that contain escaped characters (from the above class definition `ClassWithUnicodePropertyName` - line 163 `Property\u04671` ), which is throwing off all subsequent highlighting/C# validation up until this semi-colon on line 169.